### PR TITLE
v0.7 - Daric

### PR DESCRIPTION
**Pull request by AntoineRondelet**

### CODE REVIEW

**Review by dtebbs — APPROVED** [review]
LGTM.

**Comment by dtebbs** [comment]
The docker builds appear to be failing while trying to install the solidity compiler.  I can't see why it would fail, so I've restarted the job (for "pull_request") in case it's something intermittent.  If that works, we can run the "push" version too.

**Comment by dtebbs** [comment]
```
#15 830.5 python -c "from zeth.core.contracts import install_sol; \
#15 830.5 	install_sol()"
#15 832.0 Traceback (most recent call last):
#15 832.0   File "/home/zeth/client/env/lib/python3.8/site-packages/solcx/install.py", line 633, in _validate_installation
#15 832.0     installed_version = wrapper._get_solc_version(binary_path)
#15 832.0   File "/home/zeth/client/env/lib/python3.8/site-packages/solcx/wrapper.py", line 17, in _get_solc_version
#15 832.0     stdout_data = subprocess.check_output([str(solc_binary), "--version"], encoding="utf8")
#15 832.0   File "/usr/lib/python3.8/subprocess.py", line 415, in check_output
#15 832.0     return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
#15 832.0   File "/usr/lib/python3.8/subprocess.py", line 493, in run
#15 832.0     with Popen(*popenargs, **kwargs) as process:
#15 832.0   File "/usr/lib/python3.8/subprocess.py", line 858, in __init__
#15 832.0     self._execute_child(args, executable, preexec_fn, close_fds,
#15 832.0   File "/usr/lib/python3.8/subprocess.py", line 1706, in _execute_child
#15 832.0     raise child_exception_type(errno_num, err_msg, err_filename)
#15 832.0 FileNotFoundError: [Errno 2] No such file or directory: '/root/.solcx/solc-v0.8.1'
#15 832.0 
#15 832.0 During handling of the above exception, another exception occurred:
#15 832.0 
#15 832.0 Traceback (most recent call last):
#15 832.0   File "<string>", line 1, in <module>
#15 832.0   File "/home/zeth/client/zeth/core/contracts.py", line 120, in install_sol
#15 832.0     solcx.install_solc(SOL_COMPILER_VERSION)
#15 832.0   File "/home/zeth/client/env/lib/python3.8/site-packages/solcx/install.py", line 466, in install_solc
#15 832.0     raise exc
#15 832.0   File "/home/zeth/client/env/lib/python3.8/site-packages/solcx/install.py", line 459, in install_solc
#15 832.0     _validate_installation(version, solcx_binary_path)
#15 832.0   File "/home/zeth/client/env/lib/python3.8/site-packages/solcx/install.py", line 636, in _validate_installation
#15 832.0     raise SolcInstallationError(
#15 832.0 solcx.exceptions.SolcInstallationError: Downloaded binary would not execute, or returned unexpected output. If this issue persists, you can try to compile from source code using `solcx.compile_solc('0.8.1')`.
#15 832.1 make: *** [Makefile:8: setup] Error 1
#15 ERROR: executor failed running [/bin/sh -c virtualenv env     && source env/bin/activate     && pip install --no-index --find-links=wheels cryptography     && make setup]: exit code: 2
------
 > [ 9/11] RUN virtualenv env     && source env/bin/activate     && pip install --no-index --find-links=wheels cryptography     && make setup:
------
Dockerfile-client:47
--------------------
  46 |     
  47 | >>> RUN virtualenv env \
  48 | >>>     && source env/bin/activate \
  49 | >>>     && pip install --no-index --find-links=wheels cryptography \
  50 | >>>     && make setup
  51 |     
--------------------
error: failed to solve: rpc error: code = Unknown desc = executor failed running [/bin/sh -c virtualenv env     && source env/bin/activate     && pip install --no-index --find-links=wheels cryptography     && make setup]: exit code: 2
Error: Process completed with exit code 1.
```

**Comment by dtebbs** [comment]
The same command worked for me in a local docker container.  It may be something specific to the environment.

**Comment by dtebbs** [comment]
It seems solc-v0.8.1 does indeed not execute in a container based on an alpine image.  ldd reports a missing shared lib.  Investigating in #371.

**Comment by AntoineRondelet** [comment]
> It seems solc-v0.8.1 does indeed not execute in a container based on an alpine image. ldd reports a missing shared lib. Investigating in #371.

Thanks for looking into that. Yes indeed I just did a very quick test using [solc-select](https://github.com/crytic/solc-select) in an alpine:3.13 container, and the compiler seems to work fine until `0.7.5` - higher -> things break, which seems to be confirmed by looking at the alpine package registry: https://pkgs.alpinelinux.org/package/edge/testing/x86_64/solidity

File: .github/workflows/dev-docker-apps.yml [hunk 1]
```diff
@@ -32,22 +32,21 @@ jobs:
...
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
-
-      - name: "Docker build"
-        run: docker build -f ./Dockerfile-${{ matrix.app }} -t $REPO_OWNER/zeth-${{ matrix.app }}:git-${{ steps.git_vars.outputs.COMMIT_HASH }} .
+      - name: "Configure Docker"
+        run: echo ${{ secrets.CR_PAT }} | docker login ghcr.io -u $GITHUB_ACTOR --password-stdin
 
-      - name: "Docker push"
-        run: docker push $REPO_OWNER/zeth-${{ matrix.app }}:git-${{ steps.git_vars.outputs.COMMIT_HASH }}
+      - name: "Set up Buildx"
```
[the remark this answers is not in the document]
Comment (AntoineRondelet) [reply .github/workflows/dev-docker-apps.yml:38]: #372

**Pull request merged**